src/audio/module_adapter/module/waves/waves.c

```diff
@@ -456,6 +458,48 @@ static int waves_effect_revision(struct processing_module *mod)
 	return 0;
 }
 
+/* cache config blob*/
+static int waves_effect_cache_config_blob(struct processing_module *mod, void *data, uint32_t size)
+{
+	struct comp_dev *dev = mod->dev;
+	struct module_data *codec = &mod->priv;
+	struct waves_codec_data *waves_codec = codec->private;
+
+	comp_dbg(dev, "waves_effect_cache_config_blob() start");
+
+	/* release old cached config blob*/
+	if (waves_codec->config_blob && size != waves_codec->config_blob_size) {
+		comp_info(dev, "waves_effect_cache_config_blob() release blob");
+		module_free_memory(mod, waves_codec->config_blob);
+		waves_codec->config_blob = NULL;
+		waves_codec->config_blob_size = 0;
+	}
+
+	if (!waves_codec->config_blob) {
+		waves_codec->config_blob = module_allocate_memory(mod, size, 16);
+		if (!waves_codec->config_blob) {
+			comp_err(dev,
+				"waves_effect_cache_config_blob() failed to allocate %d bytes for config blob",
+				size);
+			return -ENOMEM;
+		}
+	}
+	waves_codec->config_blob_size = size;
+	int ret = memcpy_s(waves_codec->config_blob, waves_codec->config_blob_size,
+			data, size);
+	if (ret) {
+		comp_err(dev,
+			"waves_effect_cache_config_blob(): failed to copy config blob %d",
+			ret);
+		module_free_memory(mod, waves_codec->config_blob);
```

Comment:
need to set `waves_codec->config_blob` to `NULL` too

Reply:
+1 from my outdated comment https://github.com/thesofproject/sof/pull/8448#discussion_r1387899199

btw github is so not friendly to bury unaddressed comments and label them as outdated when the new-updated commit may change the relation.